test/task-affinity.jl

```diff
@@ -0,0 +1,577 @@
+                
+                @test get_compute_scope(task1) == get_compute_scope(task2) == get_compute_scope(task3) == compute_scope_intersect
+                @test get_result_scope(task1)  == get_result_scope(task2)  == get_result_scope(task3)  == chunk_scope 
+
+                execution_scope1  = get_execution_scope(task1)
+                execution_scope2  = get_execution_scope(task2)
+                execution_scope3  = get_execution_scope(task3)  
+                @test execution_scope1 == execution_scope2 == execution_scope3 == Dagger.ExactScope(chunk_proc)             
+            end
+        end
+
+        @testset "compute_scope and result_scope Tests without Intersection" begin
+            if length(availscopes) >= 2
+                n = cld(length(availscopes), 2)
+
+                shuffle!(availscopes)
+                scope_a = availscopes[1:n]
+                scope_b = availscopes[n+1:end]
+
+                compute_scope_no_intersect = Dagger.UnionScope(scope_a...)
+                scope_no_intersect         = Dagger.UnionScope(scope_a...)
+                scope_rand                 = Dagger.UnionScope(rand(availscopes, rand(1:length(availscopes))))
+                result_scope_no_intersect  = Dagger.UnionScope(scope_b...)
+
+                task1  = Dagger.@spawn compute_scope=compute_scope_no_intersect result_scope=result_scope_no_intersect Dagger.tochunk(g(10, 11), chunk_proc, chunk_scope);                  wait(task1 )
+                task2  = Dagger.@spawn scope=scope_no_intersect result_scope=result_scope_no_intersect Dagger.tochunk(g(20, 21), chunk_proc, chunk_scope);                                  wait(task2 )
+                task3  = Dagger.@spawn compute_scope=compute_scope_no_intersect scope=scope_rand result_scope=result_scope_no_intersect Dagger.tochunk(g(30, 31), chunk_proc, chunk_scope); wait(task3 )
+
+                @test get_compute_scope(task1) == get_compute_scope(task2) == get_compute_scope(task3) == compute_scope_no_intersect
+                @test get_result_scope(task1)  == get_result_scope(task2)  == get_result_scope(task3)  == Dagger.InvalidScope 
+
+                execution_scope1  = get_execution_scope(task1)
+                execution_scope2  = get_execution_scope(task2)
+                execution_scope3  = get_execution_scope(task3)  
+                @test get_execution_scope(task1) == get_execution_scope(task2) == get_execution_scope(task3) == Dagger.InvalidScope
+            end
+        end
+
+    end 
+
+    @testset "Dagger Chunk Arguments in Task: scope, compute_scope and result_scope Tests with non-Intersection of chunk arg and scope" begin 
+
+        @everywhere g(x, y) = x * 2 + y * 3
+        
+        availscopes = shuffle!(Dagger.ExactScope.([proc for i in Dagger.procs() for proc in Dagger.get_processors(Dagger.OSProc(i))]))
```

Comment:
```suggestion
        shuffle!(availscopes)
```

You can just in-place shuffle the original `availscopes`